integration_tests/sdk/delete_workflow_test.py

```diff
@@ -39,12 +40,12 @@ def test_delete_workflow_invalid_saved_objects(client, flow_name, data_integrati
     client.flow(flow.id())
 
 
-def test_delete_workflow_saved_objects(client, flow_name, data_integration, engine):
+def test_delete_workflow_saved_objects(client, flow_name, data_integration, engine, validator):
```

Comment:
I would suggest to put these as a single `context` class so that it's easier to add / change / rename all 'contexts' available to a test, without changing signatures of tests every single time.

Reply:
interesting - but that means every test is going to initialize all these fixtures, even if the test doesn't need them right?

Reply:
My understanding of the philosophy behind pytest fixtures is that they are run only when requested by the test case. Test cases that don't need to, for example, publish a flow, wouldn't need to ask for `flow_name`. And I think this improves the readability of what a test is capable of, based on just glancing at the signature, since not all test cases require all the fixtures.

I'd be willing to discuss this with a broader audience perhaps? I'm curious what others think.

Reply:
I agree with Kenny here, tests should only request fixtures that are required for the test to run.